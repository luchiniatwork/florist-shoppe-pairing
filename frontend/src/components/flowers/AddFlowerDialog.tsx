import React, { useState } from "react";
import makeStyles from "@material-ui/core/styles/makeStyles";
import {
  Button,
  Dialog,
  DialogTitle,
  Grid,
  MenuItem,
  TextField,
} from "@material-ui/core";
import { gql, useMutation, useQuery } from "@apollo/client";
import IFamily from "../../types/Family";

const useStyles = makeStyles(({ spacing }) => ({
  content: {
    padding: spacing(3),
  },
}));

interface IAddFlowerDialog {
  open: boolean;
  onClose: () => void;
  onUpdate: () => void;
}

export const GET_FAMILIES_QUERY = gql`
  query GetFamilies {
    families {
      id
      name
    }
  }
`;

export const ADD_FLOWER_MUTATION = gql`
  mutation loadMyFlower($name: String!, $familyId: ID!) {
    createFlower(flower: { name: $name, familyId: $familyId }) {
      id
      name
    }
  }
`;

const AddFlowerDialog = ({ open, onClose, onUpdate }: IAddFlowerDialog) => {
  const classes = useStyles();
  const [name, setName] = useState("");
  const [family, setFamily] = useState("");
  const { loading, data } = useQuery(GET_FAMILIES_QUERY);
  const [addFlower] = useMutation(ADD_FLOWER_MUTATION);
  return (
    <Dialog onClose={onClose} open={open}>
      <DialogTitle>Add Flower</DialogTitle>
      <div className={classes.content}>
        <Grid container spacing={3} alignContent={"center"}>
          <Grid item xs={12}>
            <TextField
              fullWidth
              label={"Name"}
              value={name}
              onChange={(e) => setName(e.target.value)}
            />
          </Grid>
          <Grid item xs={12}>
            {loading ? (
              "loading.."
            ) : (
              <TextField
                fullWidth
                label={"Family"}
                select
                value={family}
                onChange={(e) => setFamily(e.target.value)}
              >
                {data && Array.isArray(data.families)
                  ? data.families.map((family: IFamily, idx: number) => (
                      <MenuItem key={`${idx}-family`} value={String(family.id)}>
                        {family.name}
                      </MenuItem>
                    ))
                  : null}
              </TextField>
            )}
          </Grid>
          <Grid item xs={6}>
            <Button
              variant={"contained"}
              fullWidth
              color={"secondary"}
              onClick={onClose}
            >
              Cancel
            </Button>
          </Grid>
          <Grid item xs={6}>
            <Button
              variant={"contained"}
              fullWidth
              color={"primary"}
              disabled={!name || !family}
              onClick={() => {
                addFlower({ variables: { name, familyId: family } })
                  .then((res) => {
                    setName("");
                    setFamily("");
                    onUpdate();
                    onClose();
                  })
                  .catch((error) => console.log(error));
              }}
            >
              Add
            </Button>
          </Grid>
        </Grid>
      </div>
    </Dialog>
  );
};

export default AddFlowerDialog;
