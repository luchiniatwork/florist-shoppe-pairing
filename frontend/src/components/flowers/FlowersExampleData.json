[
  {
    "id": "fd55a693-b137-4c4e-8eb5-6f9f4502d141",
    "name": "Irises"
  },
  {
    "id": "69df8552-e509-402c-80a3-692abc90612a",
    "name": "Gladiolus"
  },
  {
    "id": "d5b0a4ed-ff67-42fa-87a2-cbd5191a7c30",
    "name": "Coppertips"
  },
  {
    "id": "cbe98b9d-a238-49e0-9a45-c31f552e3e43",
    "name": "Freesia"
  },
  {
    "id": "aa52dec7-cdbf-4920-9c02-9e20f2910c12",
    "name": "Bellflowers"
  },
  {
    "id": "942513f0-e864-49fe-8f97-ab7a11f97951",
    "name": "Platycodon grandiflorus"
  },
  {
    "id": "4d5ffb00-aa0e-4186-a405-0a98efb2b755",
    "name": "Lily"
  },
  {
    "id": "0c4c51f8-f91f-404e-a8f3-3a2121fe49ba",
    "name": "Tulip"
  },
  {
    "id": "066171a6-7364-4249-8909-c410902d35dd",
    "name": "Fritillaries"
  },
  {
    "id": "c7aed4c8-8fb0-458a-a950-28a789a8beea",
    "name": "Fawn lilies"
  }
]
