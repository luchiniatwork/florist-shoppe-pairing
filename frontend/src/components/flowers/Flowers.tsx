import React, { useState } from "react";
import makeStyles from "@material-ui/core/styles/makeStyles";
import {
  Button,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Paper,
} from "@material-ui/core";
import { gql, useQuery } from "@apollo/client";
import IFlower from "../../types/Flower";
import AddFlowerDialog from "./AddFlowerDialog";
import { LocalFlorist } from "@material-ui/icons";

const useStyles = makeStyles(({ spacing }) => ({
  root: {
    padding: spacing(3),
  },
}));

export const GET_FLOWERS_QUERY = gql`
  query GetFlowers {
    flowers {
      id
      name
    }
  }
`;

const Flowers = () => {
  const classes = useStyles();
  const { loading, data, refetch } = useQuery(GET_FLOWERS_QUERY);
  const [isAddDialogOpen, setIsAddDialogOpen] = useState(false);

  return (
    <Paper variant={"outlined"} className={classes.root}>
      <List>
        {!loading && data && Array.isArray(data.flowers)
          ? data.flowers.map((flower: IFlower, idx: number) => (
              <ListItem key={`${idx}-flower`}>
                <ListItemIcon>
                  <LocalFlorist />
                </ListItemIcon>
                <ListItemText primary={flower.name} />
              </ListItem>
            ))
          : "Loading..."}
        <ListItem>
          <Button
            variant={"contained"}
            onClick={() => setIsAddDialogOpen(true)}
          >
            Add Flower
          </Button>
        </ListItem>
      </List>
      <AddFlowerDialog
        open={isAddDialogOpen}
        onClose={() => setIsAddDialogOpen(false)}
        onUpdate={() => refetch()}
      />
    </Paper>
  );
};

export default Flowers;
