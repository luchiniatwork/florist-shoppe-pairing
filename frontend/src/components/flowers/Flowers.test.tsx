import React from "react";
import { render } from "react-dom";
import {act} from "react-dom/test-utils";
import { MockedProvider } from "@apollo/client/testing";
import Flowers, {GET_FLOWERS_QUERY} from "./Flowers";
import FlowersExampleData from "./FlowersExampleData.json";
import {GET_FAMILIES_QUERY} from "./AddFlowerDialog";


const mocks = [
    {
        request: {
            query: GET_FLOWERS_QUERY,
        },
        result: {data: {flowers: FlowersExampleData} },
    },
    {
        request: {
            query: GET_FAMILIES_QUERY,
        },
        result: { data: { families: [] } },
    },
];

test("Render Flowers Load State", () => {
    let container = document.createElement('div');
    act(() => {
        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Flowers />
            </MockedProvider>, container
        );
    })
    expect(container).toMatchSnapshot();
});

test("Render Flowers Final State", async () => {
    let container = document.createElement('div');
    act(() => {
        render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <Flowers />
            </MockedProvider>, container
        );
    })
    await new Promise((resolve) => setTimeout(resolve, 0)); // wait for response
    expect(container).toMatchSnapshot();
});
