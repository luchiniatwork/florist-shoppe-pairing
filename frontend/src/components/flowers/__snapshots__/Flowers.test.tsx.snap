// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Flowers Final State 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
  >
    <ul
      class="MuiList-root MuiList-padding"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Irises
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Gladiolus
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Coppertips
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Freesia
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Bellflowers
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Platycodon grandiflorus
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Lily
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Tulip
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Fritillaries
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            Fawn lilies
          </span>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add Flower
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Render Flowers Load State 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-1 MuiPaper-outlined MuiPaper-rounded"
  >
    <ul
      class="MuiList-root MuiList-padding"
    >
      Loading...
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add Flower
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </div>
</div>
`;
